File: src/main/java/com/footverse/order/entity/PaymentStatus.java
package com.footverse.order.entity;

/**
 * Payment status of an {@link Order} (dto-spec §4, database-spec §10.11). Persisted as a string via
 * {@code @Enumerated(EnumType.STRING)}. An order starts {@code UNPAID} and becomes {@code PAID} when
 * delivered.
 */
public enum PaymentStatus {

    /** Not yet paid; the initial status of every order. */
    UNPAID,

    /** Paid; set when the order is delivered. */
    PAID
}

File: app/lib/features/order/models/payment_status.dart
import 'package:json_annotation/json_annotation.dart';

/// Payment status of an order (dto-spec §4, §15). An order starts `UNPAID` and
/// becomes `PAID` when delivered.
enum PaymentStatus {
  @JsonValue('UNPAID')
  unpaid,
  @JsonValue('PAID')
  paid,
}
